components/messaging/CurrentConversation.tsx

```diff
@@ -35,10 +39,20 @@ const CurrentConversation: React.FC = () => {
   }, []);
 
   useEffect(() => {
-    dispatch({
-      type: 'SET_CURRENT_CONVERSATION',
-      payload: allConversations[0],
-    });
+    const currentConversation = allConversations.find(
+      (convo) => convo.conversation_id === conversationId
+    );
+    if (currentConversation) {
+      dispatch({
+        type: 'SET_CURRENT_CONVERSATION',
+        payload: currentConversation,
+      });
+    } else {
+      dispatch({
+        type: 'SET_CURRENT_CONVERSATION',
+        payload: allConversations[0],
+      });
+    }
```

Comment:
We would potentially not need these changes if the other adjustments work out ?

Reply:
I suggest changing it to this:
```
useEffect(() => {    
    if(!currentConversation){
      dispatch({
        type: 'SET_CURRENT_CONVERSATION',
        payload: allConversations[0],
      });
    }    
  }, [allConversations]);
```
  
  The default behavior is to install the first conversation  in the list(allConversations[0]), but it would be correct to check if it is already set currentConversation and if not, then set the first one from the list.

Reply:
Ah that evaluation would always return false I think 🤔 

The provider is set so that whenever we call a part of the context it's always guaranteed not to be `nullish` - if there were no value to `currentConversation` we would essentially crash.

So this useEffect originally I think was just meant to set it back to a default because the first conversation in the list (`allConversations[0]`) will always be the one that was most recently changed - that is defined in the order that our database returns the fetch results in.

So this useEffect will always set us to the most recent conversation which would be the new one if we have just created it - what wasn't happening before was navigating directly to the conversation instead of the list so this useEffect should be fine in it's original form

Reply:
this code worked for me, if you leave it as it is without checking currentConversation, then when opening a conversation from the item page, it will always redirect to the last conversation, which is not correct on allConversations[0], which is not what you expect. Anyway, I didn't get any errors and the right conversation opens.

Reply:
You could make this section a lot simpler by using a ternary operator:
```suggestion
    const currentConversation = allConversations.find(
      (convo) => convo.conversation_id === conversationId
    );
    dispatch({
      type: 'SET_CURRENT_CONVERSATION',
      payload: currentConversation ? currentConversation : allConversations[0],
    });
```

Reply:
And how is `allConversations` ordered? Is it by created date? Last updated date?
Picking the first element could represent different behaviours depending on how this array is ordered.

Reply:
Thanks @mnixo! At the moment, we need to change the logic a little:
if(!currentConversation){
      dispatch({
        type: 'SET_CURRENT_CONVERSATION',
        payload: allConversations[0],
      });
    } 
The idea is this: if the current conversation is already defined, then we don't need to do anything, and if the current conversation is not set, then we need to set the last created one as the current one(allConversations[0]).
But there is still room for discussion, maybe in the future we will need to adjust the logic a little :-)

Reply:
> And how is `allConversations` ordered? Is it by created date? Last updated date? Picking the first element could represent different behaviours depending on how this array is ordered.

For future note - we retrieve them so that index 0 is always the latest updated one